TrackModel/src/trackmodel/model/Block.java

```diff
@@ -547,6 +501,22 @@ public void setFailureStatus() {
     }
   }
 
+  /**
+  * This method will set the if the doors for a station are on the left side.
+  * @param leftDoors This using a boolean indicates if the doors are on the left
+  */
+  public void setLeftDoors(boolean leftDoors) {
+    this.leftStation = leftDoors;
+  }
+
+  /**
+  * This method will set the if the doors for a station are on the right side.
+  * @param rightDoors This using a boolean indicates if the doors are on the left
+  */
+  public void setRightDoors(boolean rightDoors) {
+    this.rightStation = rightDoors;
+  }
+

```

Comment:
These seem to be setting the stations, not the doors?

Reply:
Yep so that at a station the train knows which doors it can and cannot open

Reply:
That should be up to the train to figure out. The track shouldn't have instance variables for doors. You could keep the rightStation and leftStation instance variables and then the train could determine which doors to open from that.

Reply:
ie. the TrainModel would have:

if (hasStation) {
    if (leftStation) openLeftDoors();
    else openRightDoors();
}

Reply:
RightDoors is basically my way of saying rightStation, I just need to change the name.